GCC 4.9 Compile Error implicit conversion failure from initializer list

In gcc 4.9 implicit conversion from a list to unordered_map is not allowed. This is a bug in gcc 4.9 which was fixed[1] but not in the version of Ubuntu. The simple work around is instead of returning `{}` to call the unordered map constructor.

error from travis:
```
/home/travis/build/Shelnutt2/TileDB/build/tiledb/examples/cpp_api/../tiledb/query.h:360:15: error: converting to ‘std::unordered_map<std::basic_string<char>, std::pair<long unsigned int, long unsigned int> >’ from initializer list would use explicit constructor ‘std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::unordered_map(std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::size_type, const hasher&, const key_equal&, const allocator_type&) [with _Key = std::basic_string<char>; _Tp = std::pair<long unsigned int, long unsigned int>; _Hash = std::hash<std::basic_string<char> >; _Pred = std::equal_to<std::basic_string<char> >; _Alloc = std::allocator<std::pair<const std::basic_string<char>, std::pair<long unsigned int, long unsigned int> > >; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::size_type = long unsigned int; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::hasher = std::hash<std::basic_string<char> >; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::key_equal = std::equal_to<std::basic_string<char> >; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::allocator_type = std::allocator<std::pair<const std::basic_string<char>, std::pair<long unsigned int, long unsigned int> > >]’
       return {};  // Query hasn't been submitted
```

[1] https://gcc.gnu.org/bugzilla/show_bug.cgi?id=58930
[2] https://stackoverflow.com/questions/26947704/implicit-conversion-failure-from-initializer-list

--- a/tiledb/sm/cpp_api/query.h
+++ b/tiledb/sm/cpp_api/query.h
@@ -357,7 +357,9 @@ class Query {
   result_buffer_elements() const {
     std::unordered_map<std::string, std::pair<uint64_t, uint64_t>> elements;
     if (buff_sizes_.size() == 0)
-      return {};  // Query hasn't been submitted
+      return std::unordered_map<
+          std::string,
+          std::pair<uint64_t, uint64_t>>();  // Query hasn't been submitted
     for (const auto& b_it : buff_sizes_) {
       auto attr_name = b_it.first;
       auto size_pair = b_it.second;

